### PR TITLE
Fix unigram assert

### LINKED ISSUE #46898: `tf.random.fixed_unigram_candidate_sampler` crashes(abort) when `vocab_file` is invalid
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
`tf.random.fixed_unigram_candidate_sampler` crashes(abort) when `vocab_file` is invalid

**Describe the expected behavior**
expect an exception message if the file format is incorrect instead of crash

**Standalone code to reproduce the issue**
~~~python
tf.random.fixed_unigram_candidate_sampler(true_classes=np.ones((1,1)), num_true=1, num_sampled=1, unique=True, range_max=1, vocab_file='abc')
~~~
Output:
~~~python
2021-02-03 22:08:02.817419: F tensorflow/core/kernels/range_sampler.cc:246] Non-OK-status: LoadFromFile(env, vocab_file, distortion) status: Not found: abc; No such file or directory
Aborted (core dumped)
~~~

### LINKED ISSUE #58350: Check failed in fixed_unigram_candidate_sampler
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.10

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current Behaviour?

```shell
Check failed: range == weights_.size() (10 vs. 3)
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
true_classes = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
num_true = 2
num_sampled = 4
unique = True
range_max = 10
distortion = 1.0
num_reserved_ids = 0
num_shards = 1
shard = 0
unigrams = [0.1, 0.8, 0.1]
seed = None
sampler = tf.random.fixed_unigram_candidate_sampler(true_classes, num_true, num_sampled, unique, range_max, distortion=distortion, num_reserved_ids=num_reserved_ids, num_shards=num_shards, shard=shard, unigrams=unigrams, seed=seed)
```

### Relevant log output

```shell
F tensorflow/core/kernels/range_sampler.cc:264] Check failed: range == weights_.size() (10 vs. 3)
abort
```
</details>

### PR DESCRIPTION
**Pull request by bhack**
Fixes https://github.com/tensorflow/tensorflow/issues/46898

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @jpienaar Can you please review this PR ? Thank you!

**Review by mihaimaruseac — APPROVED** [review]

**Comment by jpienaar** [comment]
Thanks for addressing this and Mihai for reviewing (not sure why it was assigned to me, added Antonio for visibility)

**Review by cantonios — APPROVED** [review]

File: tensorflow/core/kernels/range_sampler_test.cc [hunk 1]
```diff
@@ -157,32 +157,54 @@ TEST_F(RangeSamplerTest, FixedUnigramProbabilities) {
...
+  string fname = "NoExistingFile";
+  FixedUnigramSampler* test_sampler = new FixedUnigramSampler(9, 0.8, 0, 1, 0);
+  Status s = test_sampler->SetDistributionSampler(env, fname);
+  EXPECT_TRUE(errors::IsNotFound(s)) << s;
+}
+TEST_F(RangeSamplerTest, FixedUnigramNoMatchingRangeWeights) {
+  Env* env = Env::Default();
+  string fname = io::JoinPath(testing::TmpDir(), "vocab_file");
+  TF_CHECK_OK(WriteStringToFile(env, fname, kVocabContent));
+  FixedUnigramSampler* test_sampler = new FixedUnigramSampler(8, 0.8, 0, 1, 0);
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/range_sampler_test.cc:179]: This causes a memleak which reverted the PR

CC @bhack

Comment (bhack) [reply tensorflow/core/kernels/range_sampler_test.cc:179]: Ok I suppose we don't have this coverage in the OSS CI?

Comment (cantonios) [reply tensorflow/core/kernels/range_sampler_test.cc:179]: I don't think we run ASAN in OSS (builds/tests take much longer).  They are run on a cadence internally.

Comment (jpienaar) [reply tensorflow/core/kernels/range_sampler_test.cc:179]: Just adding `sampler_.reset(test_sampler);` (or ensuring it is already there) here should suffice. (adding it right after the creation of the FixedUnigramSampler in each case sufficed when I just tested it)

Comment (bhack) [reply tensorflow/core/kernels/range_sampler_test.cc:179]: Blackbox edit at https://github.com/tensorflow/tensorflow/pull/59096

**Pull request merged**